test/integration/puppetlabs/services/jruby/class_info_test.clj

```diff
@@ -0,0 +1,229 @@
+                  expected-envs-info {"env1" {foo-manifest
+                                              [(expected-class-info "foo")
+                                               (expected-class-info "foo2")]}
+                                      "env2" (expected-manifests-info
+                                               [env-2-dir-and-manifests
+                                                env-1-mod-1-dir-and-manifests
+                                                env-1-mod-2-dir-and-manifests])}]
+              (testing "all environments"
+                (is (= expected-envs-info (get-class-info))))
+              (testing "one environment by name"
+                (is (= (expected-envs-info "env1")
+                       (get-class-info-for-env "env1"))
+                    "Unexpected info retrieved for 'env1'")
+                (is (= (expected-envs-info "env2")
+                       (get-class-info-for-env "env2"))
+                    "Unexpected info retrieved for 'env2'"))))
+
+          (testing "changes to manifest content for"
+            (fs/delete-dir env-1-mod-2-dir)
+            (let [foo-manifest (.getAbsolutePath (fs/file env-1-dir
+                                                          "manifests"
+                                                          "foo.pp"))
+                  _ (create-file foo-manifest "class foo () {} \n")
+                  expected-envs-info {"env1" {foo-manifest
+                                              [{"name" "foo"
+                                                "params" []}]}
+                                      "env2" (expected-manifests-info
+                                               [env-2-dir-and-manifests
+                                                env-1-mod-1-dir-and-manifests])}]
+              (testing "all environments"
+                (is (= expected-envs-info (get-class-info))))
+              (testing "one environment by name"
+                (is (= (expected-envs-info "env1")
+                       (get-class-info-for-env "env1"))
+                    "Unexpected info retrieved for 'env1'")
+                (is (= (expected-envs-info "env2")
+                       (get-class-info-for-env "env2"))
+                    "Unexpected info retrieved for 'env2'"))))
+
+          (testing "changes to environments for"
+            (fs/delete-dir env-1-dir)
+            (let [_ (create-env env-3-dir-and-manifests)
+                  expected-envs-info {"env2" (expected-manifests-info
+                                               [env-2-dir-and-manifests])
+                                      "env3" (expected-manifests-info
+                                               [env-3-dir-and-manifests
+                                                base-mod-1-and-manifests])}]
+              (testing "all environments"
+                (is (= expected-envs-info (get-class-info))))
+              (testing "one environment by name"
```

Comment:
Would it be worth trying to get the class info for `"env1"` to ensure that it won't be returned since the environment has been deleted?